src/types.rs

```diff
@@ -340,37 +341,59 @@ impl RpcAccess {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, CandidType, Serialize, Deserialize, Default)]
-pub enum LogMessageFilter {
+pub enum ConsoleFilter {
     #[default]
     ShowAll,
-    ShowOnly(Vec<LogMessageType>),
+    HideAll,
+    ShowPattern(RegexString),
+    HidePattern(RegexString),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, CandidType, Serialize, Deserialize, Default)]
+pub struct RegexString(String);
+
+impl RegexString {
+    pub fn is_match(&self, value: &str) -> bool {
+        Regex::new(&self.0)
+            .map(|regex| regex.is_match(value))
+            .unwrap_or(false)
```

Comment:
nit: I think we should panic in case the regex is not valid instead of just silently returning `false`

Reply:
I suppose we can do this because it's only used in the local replica. Updated.

Reply:
> I suppose we can do this because it's only used in the local replica.

By local replica you mean running locally in dfx? If yes, I think we can and should also do this even if we were to use that feature productively. This could only be set during an upgrade and so if the regex is wrong then the panic will revert the upgrade which is IMO the correct thing to do.

Reply:
Yes, that makes sense. This is now implemented.